Tests/OpenFeatureTests/ProviderEventsTests.swift

```diff
@@ -19,4 +19,20 @@ final class ProviderEventsTests: XCTestCase {
         wait(for: [readyExpectation], timeout: 5)
         XCTAssertNotNil(eventState)
     }
+
+    func testErrorEventSent() {
```

Comment:
Can we add a test for `setProviderAndWait` similar to [this one](https://github.com/open-feature/swift-sdk/blob/3ce6b8d7f14150c77b363df0af3ce41c0e80138d/Tests/OpenFeatureTests/DeveloperExperienceTests.swift#L52-L83)? The difference being the `.error` instead of the `.ready`